nav2_route/include/nav2_route/types.hpp

```diff
@@ -203,6 +203,15 @@ struct Route
   float route_cost{0.0};
 };
 
+struct RouteData
```

Comment:
Rename to `RouteRequest`, this is named a bit too generically. Make sure to update the variable uses from `route_data` appropriately.

Also, add doxygen for this struct